docs/pages/en-us/Error-Handling.md

```diff
@@ -0,0 +1,114 @@
+
+## ResultException
+A ResultException can be detected by the Worker as a result of some fault in the Worker logic (e.g. NullPointerException). The Worker will not crash as a result of this kind of error but the original message will never be retried.
+
+![Result Exception](images/ResultException.png)
+*Figure 4 - ResultException flow*
+
+1. The Worker will receive a message from the messaging input queue, `worker-input-queue` and will start to process it.
+2. A runtime exception is detected by the Worker.
+3. The original message is never retried but instead acknowledged.
+4. A `RESULT_EXCEPTION` response with exception details is placed on the messaging output queue `worker-output-queue` by the worker framework.
+
+## Error Handling Rules
+The general following rules should be adhered to by all Worker implementations:
+
+ - For any explicit failures, return a failure result, not an exception
+ - If the input message is not parsable, throw an InvalidTaskException
+ - If the task cannot be accepted right now, throw a TaskRejectedException
+ - If a transient error occurs in processing, throw a TaskRejectedException
+ - If the worker receives an InterruptedException, propagate the InterruptedException
+ - If a catastrophic error occurs in processing, throw TaskFailedException
+
+ The `WorkerFactory` should identify whether the task message data is
+ parsable and this is the first opportunity to throw an `InvalidTaskException`.
+ Once a Worker is created with a task object, the framework will verify the
+ object's constraints (if there are any), which is the second chance to throw
+ `InvalidTaskException`. The constructor of the `Worker` can throw an
+ `InvalidTaskException`. Finally a worker's `doWork()` method can thow an
+ `InvalidTaskException`.
+
+ While `InvalidTaskException` is a non-retryable case, there may be retryable
+ scenarios for instance a brief disconnection from a temporary resources such as
+ a database. If you have a health check in your `WorkerFactory` and this is
+ currently failing, you may wish to throw `TaskRejectedException`, which will
+ push the task back onto the queue. Once inside a Worker itself, either the
+ code or the libraries used should be able to tolerate some amount of transient
+ failures in connected resources, but if this still cannot be rectified in a
+ reasonable time frame then it is also valid to throw `TaskRejectedException` from
+ inside the Worker, with the understanding that any amount of work done so
+ far will be abandoned.
+
+ Throwing `TaskFailedException` should be a last resort, for situations that
+ should not occur and are not recoverable. Most times, any checked exceptions
+ that are caught and are a valid failure case should log the exception, (not
+ propagate the exception). Workers derived from `AbstractWorker<>` should then
+ call `createFailureResult(...)` perhaps using a sensible enumeration status code
+ for your task.
```

Comment:
I don't really understand this paragraph.  I've no idea why I would throw a `TaskFailedException` or what the behvaior of the framework would be if I did.

Reply:
OK I will drop this paragraph. I simply copied original text on the published Architecture page and moved into here.